### PR TITLE
Spacepod and Some Runtime Fixes

### PR DESCRIPTION
**Pull request by Twinmold93**
Fixes sending pod pilots to the void when the pod explodes/is destroyed, as well as giving them damage warnings, just like passengers get. Also fixes it so if a guardian traps something, then dies, doesn't runtime. Lastly, fixes it so there isn't a randomly runtime from opening a critter crate, as well as cleaning up a little snowflake code, thanks to Crazylemon.

Changes sound playing and message displaying to procs:

play_sound_to_riders(mysound)
message_to_riders(mymessage)

This clears a little bit of the duplicate code, making it a little more compressed, and also allows playing sounds directly to people in pods and messages. (Perhaps after the feature freeze, we can give pods music radios? Just a crazy idea.)

Fixes the underlying issue of #5909 where the destroyed pod deletes the pilot of the pod.

:cl:Twinmold
Fix: Pilots of spacepods will no longer be sent to the void when the space pod is destroyed in any method.
Fix: Pilots of spacepods will now get the same warnings for damage/core destruction that passengers get.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Please don't include runtimes in your changelog; they really don't mean anything to players.

File: code/modules/spacepods/spacepod.dm [hunk 1]
```diff
@@ -226,26 +229,36 @@
 	health = max(0, health - damage)
 	var/percentage = (health / initial(health)) * 100
 	occupant_sanity_check()
-	if(passengers && oldhealth > health && percentage <= 25 && percentage > 0)
+	if(passengers | pilot && oldhealth > health && percentage <= 25 && percentage > 0)
```
Comment (Crazylemon64) [inline code/modules/spacepods/spacepod.dm]: You want `||`, not `|` - you want a logical OR, not a union OR, here

File: code/modules/spacepods/spacepod.dm [hunk 2]
```diff
@@ -226,26 +229,36 @@
...
 		S.wait = 0 //No queue
 		S.channel = 0 //Any channel
 		S.volume = 50
-		for(var/mob/M in passengers)
-			M << S
-	if(passengers && oldhealth > health && !health)
+		if(pilot)
+			pilot << S
+		if(passengers)
+			for(var/mob/M in passengers)
```
Comment (Crazylemon64) [inline code/modules/spacepods/spacepod.dm]: `for(var/mob/M in passengers | pilot)`

File: code/modules/spacepods/spacepod.dm [hunk 3]
```diff
@@ -226,26 +229,36 @@
...
+				M << S
+	if(passengers | pilot && oldhealth > health && !health)
```
Comment (Crazylemon64) [inline code/modules/spacepods/spacepod.dm]: Logical OR

File: code/modules/spacepods/spacepod.dm [hunk 4]
```diff
@@ -226,26 +229,36 @@
...
 		var/sound/S = sound('sound/effects/engine_alert1.ogg')
 		S.wait = 0
 		S.channel = 0
 		S.volume = 50
-		for(var/mob/M in passengers)
-			M << S
+		if(pilot)
+			pilot << S
+		if(passengers)
+			for(var/mob/M in passengers)
```
Comment (Crazylemon64) [inline code/modules/spacepods/spacepod.dm]: If you're doing this so often, perhaps you might want a proc to play a sound to all occupants of a pod, as well as a proc to display a message to all occupants, as well.

**Comment by Twinmold93** [comment]
Requested changes needing to be reviewed.

**Pull request merged**